android/src/main/java/com/reactnativecommunity/netinfo/NetInfoModule.java

```diff
@@ -63,10 +64,14 @@ public void onAmazonFireDeviceConnectivityChanged(boolean isConnected) {
     @ReactMethod
     public void addListener(String eventName) {
         // Keep: Required for RN built in Event Emitter Calls.
+        mConnectivityReceiver.hasListener = true;
     }
 
     @ReactMethod
     public void removeListeners(Integer count) {
         // Keep: Required for RN built in Event Emitter Calls.
+        if (count == 0) {
+            mConnectivityReceiver.hasListener = false;
+        }
```

Comment:
I think count is the number of listeners to remove, so we need to track the number of listener and if the number of listeners is 0 then hasListener should be false

Reply:
nice catch! I peeled it to a new issue for tracking